### ISSUE TITLE
Events

### ISSUE THREAD

**Issue by WeTurkstra** [LYDIC-GROUP/rapid-api-crud-bundle#14]
To add the ability to change certain behaviour we want to add a whole bunch of events.
All controller actions should contain events. 

Events to add:
PreCreateEntityEvent.
PostCreateEntityEvent. 
PreUpdateEntityEvent,
PostUpdateEntityEvent,
PreDeleteEntityEvent,
PostDeleteEntityEvent,
PreListEntitiesEvent,
PostListEntitiesEvent,
PreFindEntityEvent,
PostFindEntityEvent

All these event should get the following parameters:
-The Context.
-The subject of the action. Ex. Entity, Entities or queryBuilder.

In addition to this i think the serializer groups should be included in the controllerConfig. So they can be modified using events.
This way we can manipulate the output even more! This might be a separate issue 😇 

Please add your thoughts in the comments below 😄

**Comment by Rellix999** [comment]
Yes, great addition. I also like the idea of adding serializerGroups to the config! More flexibility.

Two more events:
PreFindEntity
PostFindEntity

**Comment by WeTurkstra** [comment]
@Rellix999 Added events to the list
